### PR TITLE
build(package-lock): updated dependencies

### PR DESCRIPTION
**Pull request by rrieger-espirit**
updated all deprecated dependencies to the latest version

### CODE REVIEW

**Comment by dbarczewski** [comment]
Code review looks fine to me. 
I will need to QA this later. 
For now, could you please resolve the merge conflict?

**Comment by dbarczewski** [comment]
I could not start the examples with `npm run serve` or start the build process `npm run build` because there are several errors.
Please check them out.

File: .eslintrc.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ module.exports = {
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/no-explicit-any": "off",
     "@typescript-eslint/no-non-null-assertion": "off",
+    "@typescript-eslint/ban-ts-ignore": "off",
```
Comment (henczi-espirit) [inline .eslintrc.js:24]: why is it needed? Isn't it better to activate it?

File: .eslintrc.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ module.exports = {
...
+    "@typescript-eslint/explicit-module-boundary-types": "off",
```
Comment (henczi-espirit) [inline .eslintrc.js:25]: why is it needed? Isn't it better to activate it?

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
```
Comment (henczi-espirit) [inline .husky/pre-commit:2]: What is it doing?

File: examples/tsx/index.tsx [hunk 4]
```diff
@@ -13,7 +13,7 @@ import ListItem from "./components/richtext/ListItem";
 @Component({
   name: "App",
 })
-class App extends TsxComponent<{}> {
+class App extends TsxComponent<unknown> {
```
Comment (henczi-espirit) [inline examples/tsx/index.tsx:16]: What is "unknown" in this context?

File: package.json [hunk 5]
```diff
@@ -27,65 +28,66 @@
     "README.md"
   ],
   "dependencies": {
-    "core-js": "^2.6.10",
-    "date-fns": "^2.14.0",
-    "fsxa-api": "^5.0.0",
+    "core-js": "^3.15.2",
+    "date-fns": "^2.22.1",
+    "fsxa-api": "^5.3.0",
```
Comment (henczi-espirit) [inline package.json:33]: Isn't v 5.3.0 the version with the deprecated dependencies?

File: package.json [hunk 6]
```diff
@@ -27,65 +28,66 @@
...
+    "prettier": "^2.3.2",
     "rimraf": "^3.0.2",
-    "sass-loader": "^8.0.2",
-    "semantic-release": "^17.1.1",
-    "tailwindcss": "^1.9.6",
-    "typescript": "~3.9.3",
-    "vue-template-compiler": "^2.6.11"
+    "sass-loader": "^12.1.0",
+    "semantic-release": "^17.4.4",
+    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.2.4",
```
Comment (henczi-espirit) [inline package.json:85]: what does this complex version number mean?

File: src/components/Layout.tsx [hunk 7]
```diff
@@ -25,7 +25,7 @@ export interface LayoutProps<Data, Meta> {
 @Component({
   name: "Layout",
 })
-class Layout<Data = {}, Meta = {}> extends RenderUtils<
+class Layout<Data = unknown, Meta = unknown> extends RenderUtils<
```
Comment (henczi-espirit) [inline src/components/Layout.tsx:28]: What is "unknown" in this context?

File: src/components/RichText/index.tsx [hunk 8]
```diff
@@ -14,7 +14,11 @@ import TabbedContent from "@/components/internal/TabbedContent";
 @Component({
   name: "FSXARichText",
 })
-class RichText extends BaseComponent<RichTextProps, {}, Record<string, any>> {
+class RichText extends BaseComponent<
+  RichTextProps,
+  unknown,
```
Comment (henczi-espirit) [inline src/components/RichText/index.tsx:19]: What is "unknown" in this context?

File: src/components/base/BaseComponent.tsx [hunk 9]
```diff
@@ -25,9 +25,9 @@ import { determineCurrentRoute } from "@/utils/navigation";
   name: "BaseComponent",
 })
 class BaseComponent<
-  Props = {},
-  EventsWithOn = {},
-  Slots = {}
+  Props = unknown,
```
Comment (henczi-espirit) [inline src/components/base/BaseComponent.tsx:28]: What is "unknown" in this context?

File: src/components/base/BaseLayout.tsx [hunk 10]
```diff
@@ -13,10 +13,10 @@ import RenderUtils from "./RenderUtils";
   name: "FSXABaseLayout",
 })
 class BaseLayout<
-  Data = {},
-  Meta = {},
-  EventsWithOn = {},
-  Slots = {}
+  Data = unknown,
```
Comment (henczi-espirit) [inline src/components/base/BaseLayout.tsx:16]: What is "unknown" in this context?

File: src/components/base/BaseSection.tsx [hunk 11]
```diff
@@ -5,10 +5,10 @@ import RenderUtils from "./RenderUtils";
 class BaseSection<
   Payload = Record<string, any>,
   Meta = Record<string, any>,
-  EventsWithOn = {},
+  EventsWithOn = unknown,
```
Comment (henczi-espirit) [inline src/components/base/BaseSection.tsx:8]: What is "unknown" in this context?

File: src/components/base/RenderUtils.tsx [hunk 12]
```diff
@@ -7,9 +7,9 @@ import BaseComponent from "./BaseComponent";
   name: "RenderUtils",
 })
 class RenderUtils<
-  Props = {},
-  EventsWithOn = {},
-  Slots = {}
+  Props = unknown,
```
Comment (henczi-espirit) [inline src/components/base/RenderUtils.tsx:10]: What is "unknown" in this context?

File: src/components/internal/PortalProvider.tsx [hunk 13]
```diff
@@ -6,7 +6,7 @@ import { FSXA_INJECT_KEY_SET_PORTAL_CONTENT } from "@/constants";
 @Component({
   name: "PortalProvider",
 })
-class PortalProvider extends TsxComponent<{}> {
+class PortalProvider extends TsxComponent<unknown> {
```
Comment (henczi-espirit) [inline src/components/internal/PortalProvider.tsx:9]: What is "unknown" in this context?

File: src/shims-ts.d.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable @typescript-eslint/no-empty-interface */
+import Vue, { VNode } from "vue";
+
+declare global {
```
Comment (henczi-espirit) [inline src/shims-ts.d.ts:4]: Why is it needed

File: src/types/components.d.ts [hunk 15]
```diff
@@ -11,9 +11,9 @@ import { FSXAAppState, FSXAAppError } from "./../store";
 import { GCAPage } from "fsxa-api";
 
 export class FSXABaseComponent<
-  Props = {},
-  EventsWithOn = {},
-  Slots = {}
+  Props = unknown,
```
Comment (henczi-espirit) [inline src/types/components.d.ts:14]: What is "unknown" in this context?

File: src/types/components.d.ts [hunk 16]
```diff
@@ -86,7 +86,7 @@ export class FSXABaseComponent<
   setStoredItem<Type = any>(key: string, value: Type, ttl?: number): void;
 }
 
-export interface BaseLayoutProps<Data = {}, Meta = {}> {
+export interface BaseLayoutProps<Data = unknown, Meta = unknown> {
```
Comment (henczi-espirit) [inline src/types/components.d.ts:89]: What is "unknown" in this context?

File: src/types/components.d.ts [hunk 17]
```diff
@@ -146,8 +145,8 @@ export interface BaseSectionProps<Payload, Meta> {
 export class FSXABaseSection<
   Payload = Record<string, any>,
   Meta = Record<string, any>,
-  EventsWithOn = {},
-  Slots = {}
+  EventsWithOn = unknown,
```
Comment (henczi-espirit) [inline src/types/components.d.ts:148]: What is "unknown" in this context?

File: src/types/components.d.ts [hunk 18]
```diff
@@ -195,14 +194,14 @@ export interface BaseAppLayoutProps {
...
-  {},
+  unknown,
   {
     /**
      * This is the content that should be displayed inside of your AppLayout
      *
      * This already contains the automatically mapped layouts and sections so make sure that you embed it
      */
-    default: {};
+    default: unknown;
```
Comment (henczi-espirit) [inline src/types/components.d.ts:204]: What is "unknown" in this context?

File: src/types/components.d.ts [hunk 19]
```diff
@@ -336,7 +335,7 @@ export interface BaseRichTextElementProps<Data = Record<string, any>> {
   data: Data;
 }
 export class FSXABaseRichTextElement<
-  Data = Record<string, any>
+  Data = Record<string, any>,
```
Comment (henczi-espirit) [inline src/types/components.d.ts:338]: Why is this comma needed?

File: .eslintrc.js (hunk 1 above)
Comment (henczi-espirit) [reply .eslintrc.js:24]: sometimes this is needed

File: .eslintrc.js (hunk 2 above)
Comment (henczi-espirit) [reply .eslintrc.js:25]: sometimes this is needed

File: .husky/pre-commit (hunk 3 above)
Comment (henczi-espirit) [reply .husky/pre-commit:2]: see in file

File: package.json (hunk 5 above)
Comment (henczi-espirit) [reply package.json:33]: first release api. then update this dependency here and release this component

File: package.json (hunk 6 above)
Comment (henczi-espirit) [reply package.json:85]: compatibility mode to postcss7

File: package.json (hunk 5 above)
Comment (henczi-espirit) [reply package.json:33]: also @dbarczewski remove this dependency

File: src/main.ts [hunk 20]
```diff
@@ -1,7 +1,7 @@
 import "./tailwind.css";
 import Vue from "vue";
 import TsxApp from "./../examples/tsx";
-// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (henczi-espirit) [inline src/main.ts:4]: @dbarczewski  todo clarify

File: src/shims-ts.d.ts (hunk 14 above)
Comment (henczi-espirit) [reply src/shims-ts.d.ts:4]: needed by Typescript

**Comment by dbarczewski** [comment]
Dependencies were updated in #15. I will close this PR.

**Pull request closed without merging**